api/src/main/java/org/openmrs/api/impl/OrderSetServiceImpl.java

```diff
@@ -0,0 +1,109 @@
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.impl;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.openmrs.OrderSet;
+import org.openmrs.OrderSetMember;
+import org.openmrs.api.APIException;
+import org.openmrs.api.OrderSetService;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.db.OrderSetDAO;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.StringUtils;
+
+import java.util.Date;
+import java.util.List;
+
+@Transactional
+public class OrderSetServiceImpl extends BaseOpenmrsService implements OrderSetService {
+	
+	protected OrderSetDAO dao;
+	
+	/**
+	 * @see org.openmrs.api.OrderSetService#setOrderSetDAO(org.openmrs.api.db.OrderSetDAO)
+	 */
+	public void setOrderSetDAO(OrderSetDAO dao) {
+		this.dao = dao;
+	}
+	
+	@Override
+	public OrderSet save(OrderSet orderSet) throws APIException {
+		return saveOrderSetInternal(orderSet);
+	}
+
+	/**
+	 * @see org.openmrs.api.OrderSetService#retireOrderSet(OrderSet, String)
+	 */
+	@Override
+	public OrderSet retireOrderSet(OrderSet orderSet, String retireReason) throws APIException {
+		if (!StringUtils.hasLength(retireReason)) {
+			throw new IllegalArgumentException("reitreReason cannot be empty or null");
+		}
+		return saveOrderSetInternal(orderSet);
+	}
+	
+	/**
+	 * @see org.openmrs.api.OrderSetService#unRetireOrderSet(OrderSet, String)
+	 */
+	@Override
+	public OrderSet unRetireOrderSet(OrderSet orderSet, String unRetireReason) throws APIException {
+		return saveOrderSetInternal(orderSet);
+	}
+	
+	/**
+	 * @see org.openmrs.api.OrderSetService#save(OrderSet)
+	 */
+	private synchronized OrderSet saveOrderSetInternal(OrderSet orderSet) throws APIException {
+		if (CollectionUtils.isNotEmpty(orderSet.getOrderSetMembers())) {
+			for (OrderSetMember orderSetMember : orderSet.getOrderSetMembers()) {
+				if (orderSetMember.isVoided()) {
+					orderSetMember.setVoidedBy(Context.getAuthenticatedUser());
+					orderSetMember.setDateVoided(new Date());
+				}
+			}
+		}
+		return dao.save(orderSet);
+	}
+	
+	/**
+	 * @see org.openmrs.api.OrderSetService#getOrderSets(boolean)
+	 */
+	@Override
+	public List<OrderSet> getOrderSets(boolean includeRetired) throws APIException {
+		return dao.getOrderSets(includeRetired);
+	}
+	
+	/**
+	 * @see org.openmrs.api.OrderSetService#getOrderSetById(Integer)
+	 */
+	@Override
+	public OrderSet getOrderSetById(Integer orderSetId) throws APIException {
+		return dao.getOrderSetById(orderSetId);
+	}
+	
+	/**
+	 * @see org.openmrs.api.OrderSetService#getOrderSetByUuid(String)
+	 */
+	@Override
+	public OrderSet getOrderSetByUuid(String orderSetUuid) throws APIException {
+		return dao.getOrderSetByUniqueUuid(orderSetUuid);
+	}
+	
+	@Override
```

Comment:
I think that since you're extending BaseOpenmrsService you don't need to implement these empty methods.